### ISSUE TITLE
Potential discrepancy between claimed method and implementation

### ISSUE THREAD

**Issue by qzzxze** [sand-nine/ADG#1]
Thank you for providing the code. I obtained it from the supplementary material on OpenReview, as the GitHub repository linked in the paper (`https://github.com/sand-nine/ADG`) is currently empty. While reviewing the code, I noticed a potential inconsistency with the description in the paper.

In the paper, the authors claim:

> “Our approach recovers clean data purely from the corrupted dataset, without requiring any external information or supervision.”

However, in the code (excerpt below), it appears that the detected noisy samples are filtered using ground‑truth attack labels (`attack_idx`), which indicates prior knowledge of which samples were actually corrupted:

```python
detected_noised_indexes = torch.where(pred_noise_list[k] > threshold)[0]
detected_clean_indexes  = torch.where(pred_noise_list[k] <= threshold)[0]

for dict_data in [filted_data, filted_true_data]:
    dict_data["a"].append(self.action[k][detected_clean_indexes])
    dict_data["ns"].append(self.next_state[k][detected_clean_indexes])
    dict_data["r"].append(self.reward[k][detected_clean_indexes])
    # Force Terminal
    self.terminal[k][-1] = 1
    dict_data["t"].append(self.terminal[k][detected_clean_indexes])
    dict_data["w"].append(pred_noise_list[k][detected_clean_indexes])

the_attack_idx = torch.where(attack_idx==1)[0].cuda()

# Cheat
detected_noised_indexes = detected_noised_indexes[torch.isin(detected_noised_indexes, the_attack_idx)]
```

Specifically, `detected_noised_indexes` is intersected with `the_attack_idx` (the ground‑truth indices of attacked samples) before being used further. This effectively restricts the cleaning process to samples that are both predicted as noisy *and* truly attacked.

If my understanding is correct, this means the method relies on oracle information (i.e., knowledge of which samples were actually attacked), contradicting the claim of being fully unsupervised. Additionally, the comment `# Cheat` in the code raises further concerns about whether the implementation aligns with the described approach.

Could you please clarify whether this part of the code indeed uses ground‑truth labels, and if so, how this reconciles with the claim of unsupervised recovery? Also, given that the GitHub repository remains empty, could you provide the complete, final code as promised?

Thank you.
